probe_src/python/probe_py/manual/cli.py

```diff
@@ -146,5 +195,539 @@ def dump(
+                scp_command = [
+                    "scp",
+                    "-P",
+                    str(port),  # Specify the port
+                    f"{user_name_and_ip_src}:{process_by_id_src_remote}",
+                    f"{user_name_and_ip_dest}:{process_by_id_dest_remote}",
+                ]
+
+                upload_files(scp_command)
+            create_remote_dest_inode_and_process(src_inode_version, src_inode_metadata, remote_home_dest, dest_inode_version, dest_inode_metadata, dest_host, dest_user, port, cmd)
+        else:
+            src_path = pathlib.Path(source)
+            dest_path = pathlib.Path(os.path.join(destination, os.path.basename(src_path)))
+            src_inode_version, src_inode_metadata = get_file_info_on_local(src_path)
+            cmd.insert(0, f"-P {port}")
+            cmd.insert(0, "scp")
+            upload_files(cmd)
+            dest_inode_version, dest_inode_metadata = get_file_info_on_local(dest_path)
+            process_closure, inode_version_writes = get_prov_upstream(src_inode_version, "local")
+
+            process_id_path = (
+                    PROCESS_ID_THAT_WROTE_INODE_VERSION / src_inode_version.str_id()
+            )
+            if process_id_path.exists():
+
+                scp_command = [
+                    "scp",
+                    "-P",
+                    str(port),
+                    str(process_id_path),
+                    str(PROCESS_ID_THAT_WROTE_INODE_VERSION),  # Remote file path
+                ]
+                upload_files(scp_command)
+
+            #  transfer process from src to dest
+            process_id = inode_version_writes[src_inode_version]
+            if process_id is not None:
+                process_path = PROCESSES_BY_ID / str(process_id)
+                if process_path.exists():
+                    scp_command = [
+                        "scp",
+                        "-P",
+                        str(port),
+                        str(process_path),
+                        str(PROCESSES_BY_ID),
+                    ]
+                    upload_files(scp_command)
+
+            create_local_dest_inode_and_process(src_inode_version, src_inode_metadata, dest_inode_version, dest_inode_metadata, cmd)
+    except Exception as e:
+        traceback.print_exc()
+        print(str(e))
+
+
+def get_host_and_user(user_name_and_ip:str)->tuple[str, str]:
+    host = user_name_and_ip.split("@")[1]
+    user = user_name_and_ip.split("@")[0]
+    return host, user
+
+def create_directories_on_remote(remote_home:pathlib.Path, remote_user:str, remote_host:str)->None:
+    remote_directories = [
+        f"{remote_home}/processes_by_id",
+        f"{remote_home}/process_id_that_wrote_inode_version",  # Add more directories as needed
+    ]
+    for directory in remote_directories:
+        mkdir_command = [
+            "ssh",
+            "-p",
+            "2222",
```

Comment:
Since we only see the [`scp` arguments](https://www.man7.org/linux/man-pages/man1/scp.1.html), we should "translate" them into [`ssh` arguments](https://www.man7.org/linux/man-pages/man1/ssh.1.html). E.g., `-4`, `-6`, `-A`, `-C`, ... appear to do the same thing in scp as in ssh. However, `-P`, ... in scp does the same thing as `-p` in ssh. Finally, `-3`, `-B`, ... are only relevant for SCP not for SSH.

If not found after a quick search, we can assume the option belongs to the third category.